pkg/dao/dao.go

```diff
@@ -63,6 +67,34 @@ func (d *Dao) SetRaw(key string, val string) error {
 	return err
 }
 
+func (d *Dao) GetEtcdVersion(config Config) (string, string, error) {
+
+	// The next etcd release (1.4) will have client.GetVersion()
+	// We'll use this to test our etcd connection for now
+	resp, err := http.Get("http://" + config.EtcdHost + ":" + config.EtcdPort + "/version")
+	if err != nil {
+		return "", "", err
+	}
+
+	defer resp.Body.Close()
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		var vresp version.Versions
+		if err := json.Unmarshal(body, &vresp); err != nil {
+			return "", "", err
+		}
+		return vresp.Server, vresp.Cluster, nil
+	default:
+		var connectErr error
+		if err := json.Unmarshal(body, &connectErr); err != nil {
+			return "", "", err
+		}
+		return "", "", connectErr
+	}
+}
```

Comment:
This is cool; so we're just confirming a connection here, not validating a version range?